### ISSUE TITLE
pmdaroot crash in docker_name_matching

### ISSUE THREAD

**Issue by brendangregg** [performancecopilot/pcp#333]
Using a pmdaroot build with CFLAGS=-g:

```
$ gdb /var/lib/pcp/pmdas/root/pmdaroot pmdaroot.core.7*
...
Program terminated with signal SIGSEGV, Segmentation fault.
#0  __strcmp_sse2_unaligned () at ../sysdeps/x86_64/multiarch/strcmp-sse2-unaligned.S:32
32	../sysdeps/x86_64/multiarch/strcmp-sse2-unaligned.S: No such file or directory.
(gdb) bt
#0  __strcmp_sse2_unaligned () at ../sysdeps/x86_64/multiarch/strcmp-sse2-unaligned.S:32
#1  0x00000000004049c8 in docker_name_matching (dp=<optimized out>, query=0x7ffc0b85f790 "Titus-9707733-worker-0-255", username=0x0, 
    instname=0xa54d00 "92960bb4d157137c4eed2f33997bb61fb2d6156c365a6a0276c5b384b90dd411") at docker.c:290
#2  0x0000000000403cbe in root_container_search (query=query@entry=0x7ffc0b85f790 "Titus-9707733-worker-0-255") at root.c:165
#3  0x0000000000402d5d in root_processid_request (pdulen=<optimized out>, pdu=0x607820 <buffer>, cp=0xa4a6fc) at root.c:575
#4  root_handle_client_input (fds=<optimized out>) at root.c:728
#5  root_main (dp=0x7ffc0b85d660) at root.c:782
#6  main (argc=<optimized out>, argv=<optimized out>) at root.c:873
```

Which does:

```C
docker_name_matching(struct container_engine *dp, const char *query,
    const char *username, const char *instname)
{
    unsigned int ilength, qlength, limit;
    int i, fuzzy = 0;

    if (strcmp(query, username) == 0)
    return 100;
    if (username[0] == '/' && strcmp(query, username+1) == 0)
[...]
```

But username=0x0.

**Comment by natoscott** [comment]
Hi Brendan,

Would it to be possible to get a tarball containing /var/lib/docker/containers/*/config.* from a machine where the failure happens?  (maybe sent privately, if it contains sensitive configuration information from your environment)

cheers.

**Comment by natoscott** [comment]
Although I've been unable to reproduce this, I traced the code and attempted a fix via commit 7567e4fa (at the very least the problem will morph, as there is additional defensive code there as well now).

Please let me know if you see the problem again @brendangregg ?  Thanks!
